x-pack/plugins/transform/public/app/hooks/use_index_data.ts

```diff
@@ -101,7 +103,11 @@ export const useIndexData = (
       },
     };
 
-    const resp = await api.esSearch(esSearchRequest);
+    const { rawResponse: resp } = await lastValueFrom(
+      data.search.search({
+        params: esSearchRequest,
+      })
+    );
```

Comment:
nit: to avoid `lastValueFrom` repetition and extracting `rawResponse` I reckon it's worth creating `useDataSearch` hook that takes care of it. 

```typescript
const dataSearch = useDataSearch();

const response  = await dataSearch({ params: esSearchRequest })
```

Reply:
Added the custom hook, also added support for abort signals. Once we do the same for the `ml` plugin, we can think of moving that hook to a shared place so both plugins can make use of it. The code is based on this example: https://github.com/elastic/kibana/blob/main/examples/controls_example/public/search_example.tsx